orcid-core/src/main/java/org/orcid/core/web/filters/CrossDomainWebManger.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * ORCID (R) Open Source
+ * http://orcid.org
+ *
+ * Copyright (c) 2012-2014 ORCID, Inc.
+ * Licensed under an MIT-Style License (MIT)
+ * http://orcid.org/open-source-license
+ *
+ * This copyright and license information (including a link to the full license)
+ * shall be included in its entirety in all copies or substantial portion of
+ * the software.
+ *
+ * =============================================================================
+ */
+package org.orcid.core.web.filters;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.orcid.pojo.ajaxForm.PojoUtil;
+import org.springframework.beans.factory.annotation.Value;
+
+public class CrossDomainWebManger {
+
+    private static final String LOCALHOST = "localhost";
+    
+    @Value("${org.orcid.security.cors.allowed_domains:orcid.org}")
+    private String allowedDomains;
+    
+    @Value("${org.orcid.security.cors.allowed_path:/public}")
+    private String allowedPaths;
+    
+    @Value("${org.orcid.security.cors.allowed_functions}")
+    private String allowedFunctions;    
+    
+    private List<String> domainsRegex;  
+    
+    public boolean allowed(HttpServletRequest request) throws MalformedURLException {
+        URL url = new URL(request.getRequestURL().toString());
+        String path = url.getPath();        
+        
+        //Check origin header
+        if(!PojoUtil.isEmpty(request.getHeader("origin"))) {
+            if (validateDomain(request.getHeader("origin"))) {
+                return true;
+            }
+        } else {
+            //Check referer header for localhost
+            if(!PojoUtil.isEmpty(request.getHeader("referer"))) {
+                URL netUrl = new URL(request.getHeader("referer"));
+                String domain = netUrl.getHost();
+                if (LOCALHOST.equals(domain)) {
+                    return true;
+                }
+            }
+        }                       
+        
+        if(validatePath(path)) {
+            return true;
+        }
+        
+        if(validateFunction(path)) {
+            return true;
+        }
+        
+        return false;
+    }
+
+    private boolean validateDomain(String url) throws MalformedURLException {
+        URL netUrl = new URL(url);
+        String domain = netUrl.getHost();
+        for (String allowedDomain : getAllowedDomainsRegex()) {
+            if (domain.matches(allowedDomain)) {
+                return true;
+            }
+        }
+        return false;
+    }    
+    
+    private boolean validatePath(String path) {
+        for(String allowedPath : allowedPaths.split(",")) {
+            if(path.startsWith("/" + allowedPath)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private boolean validateFunction(String path) {
```

Comment:
Well, so far we have the public path, which will work for 

public_widgets
/public/group/{groupId}

Do you want me to restrict it more? Like allowing only public_widgets and /public/